src/main/java/com/exactpro/th2/common/schema/grpc/router/impl/DefaultGrpcRouter.java

```diff
@@ -61,7 +65,21 @@
      * @throws ClassNotFoundException if matching the service class to protobuf stub has failed
      */
     public <T> T getService(@NotNull Class<T> cls) throws ClassNotFoundException {
-        return getProxyService(cls);
+        Objects.requireNonNull(cls, "Services class can not be null");
+
+        TH2Impl implementationAnnotation = cls.getAnnotation(TH2Impl.class);
+
+        if (implementationAnnotation == null) {
+            return getProxyService(cls);
+        }
+
+        Class<?> th2ImplClass = implementationAnnotation.value();
+
+        try {
+            return (T) th2ImplClass.getConstructor(RetryPolicy.class, StubStorage.class).newInstance(configuration.getRetryConfiguration(), stubsStorages.computeIfAbsent(cls, key -> new DefaultStubStorage<>(getServiceConfig(key))));
```

Comment:
I think the line is too long. We should split it into several separate lines (extract variables or call methods in chain each one on the new line)